internal/timecmp/assert.go

```diff
@@ -16,12 +17,27 @@ type stringableTimeValue interface {
 //
 // This simplifies comparing Go stdlib time.Time values with apiserver metav1.Time / metav1.MicroTime values
 // based on the minimum granularity between the two values.
-func AssertTimeEqual(t testing.TB, expected stringableTimeValue, actual stringableTimeValue) bool {
+func AssertTimeEqual(t testing.TB, expected stringableTimeValue, actual stringableTimeValue) (equal bool) {
 	t.Helper()
+
+	defer func() {
+		if !equal {
+			assert.Fail(t, fmt.Sprintf("Not equal: \n"+
+				"expected: %s\n"+
+				"actual  : %s", expected, actual))
+		}
+	}()
+
+	expectedNil := isNilValue(expected)
+	actualNil := isNilValue(actual)
+	if expectedNil && actualNil {
```

Comment:
`expectedNil == actualNil`?

Reply:
Hmmm, I don't think so? If they're both non-nil, that'd be `true`, in which case we want to proceed with the `Equal` call that will normalize them by granularity before comparison.

FWIW I'm actually about to open another PR that will make the real comparison functions work with `nil`, at which point this code will go away.

Reply:
oops, you're right, i misread smth